### PR TITLE
TSAGE: Add keymapper support

### PR DESCRIPTION
**Pull request by NabeelShabbir**

### CODE REVIEW

File: engines/tsage/metaengine.cpp [hunk 1]
```diff
@@ -154,10 +159,186 @@ class TSageMetaEngine : public AdvancedMetaEngine<TsAGE::tSageGameDescription> {
...
 };
 
+Common::KeymapArray TSageMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace TsAGE;
+
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "tsage-default", _("Default keymappings"));
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game keymappings"));
+	Keymap *blueForceKeyMap = new Keymap(Keymap::kKeymapTypeGame, "blueforce-shortcuts", _("Blue force keymappings"));
+	Keymap *ringworldKeyMap = new Keymap(Keymap::kKeymapTypeGame, "ringworld-shortcuts", _("Ringworld keymappings"));
```
Comment (OMGPizzaGuy) [inline engines/tsage/metaengine.cpp:172]: The blueForceKeyMap and ringworldKeyMap do not appear to have assigned actions below, and they're not in the returned keymap arrray

Comment (bluegr) [reply engines/tsage/metaengine.cpp]: @OMGPizzaGuy fixed

**Comment by bluegr** [comment]
@OMGPizzaGuy as @NabeelShabbir seems to not have free time to work on this anymore, I've fixed and cleaned up the keymap code. I've played Ringworld, Ringworld 2 and Blue Force, and all seem to be working well.

I'm marking this as ready for review

**Review by OMGPizzaGuy — APPROVED** [review]
Looks good to me!

**Comment by bluegr** [comment]
Great, thanks! Squashing

**Pull request merged**